tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -50,82 +50,93 @@ astDeclarationsForLanguage language filePath = do
 syntaxDatatype :: Ptr TS.Language -> Datatype -> Q [Dec]
 syntaxDatatype language datatype = case datatype of
   SumType (DatatypeName datatypeName) _ subtypes -> do
-    cons <- traverse (constructorForSumChoice datatypeName) subtypes
-    result <- symbolMatchingInstanceForSums language name subtypes
-    pure $ generatedDatatype name cons:result
+    typeParameterName <- newName "a"
+    cons <- traverse (constructorForSumChoice datatypeName typeParameterName) subtypes
+    result <- symbolMatchingInstanceForSums language name subtypes typeParameterName
+    pure $ generatedDatatype name cons typeParameterName:result
   ProductType (DatatypeName datatypeName) _ children fields -> do
-    con <- ctorForProductType datatypeName children fields
-    result <- symbolMatchingInstance language name datatypeName
-    pure $ generatedDatatype name [con]:result
+    typeParameterName <- newName "a"
+    con <- ctorForProductType datatypeName typeParameterName children fields
+    result <- symbolMatchingInstance language name datatypeName typeParameterName
+    pure $ generatedDatatype name [con] typeParameterName:result
   LeafType (DatatypeName datatypeName) named -> do
-    con <- ctorForLeafType named (DatatypeName datatypeName)
-    result <- symbolMatchingInstance language name datatypeName
+    typeParameterName <- newName "a"
+    con <- ctorForLeafType named (DatatypeName datatypeName) typeParameterName
+    result <- symbolMatchingInstance language name datatypeName typeParameterName
     pure $ case named of
-      Anonymous -> generatedDatatype name [con]:result
-      Named -> NewtypeD [] name [] Nothing con deriveClause:result
+      Anonymous -> generatedDatatype name [con] typeParameterName:result
```

Comment:
Could/should this case construct a `newtype` now that it has a single field?

Reply:
Don't think so? 

```
The constructor of a newtype must have exactly one field
        but ‘AnonymousAwait’ has two
```

Reply:
Oh, interesting! I’m surprised—I would have thought it would just have the annotation field.

Reply:
Answering my own question, it looks like it’s [also including the bytes field](https://github.com/tree-sitter/haskell-tree-sitter/pull/181#discussion_r322294658), tho I’m [not sure if that’s a regression or not](https://github.com/tree-sitter/haskell-tree-sitter/pull/181#issuecomment-529541389).